core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -4127,14 +4130,14 @@ private static class LogicalNotImplementor extends AbstractRexCallImplementor {
    * <p>Handles all logarithm functions using log rules to determine the
    * appropriate base (i.e. base e for LN).
    */
-  private static class LogImplementor extends AbstractRexCallImplementor {
-    LogImplementor() {
-      super("log", NullPolicy.STRICT, true);
+  private static class LogImplementor extends MethodImplementor {
+    LogImplementor(Method method) {
+      super(method, NullPolicy.STRICT, true);
```

Comment:
shouldn't we be passing a String here? Why change the class we're extending?

Reply:
@tanclary I hope you can listen to my analysis here, because the functions of log and log_mysql_spark in sqlfunctions are different. If it is passed in through a string, I also need to judge whether the string is log or log_mysql_spark. If it is passed through the method method, I don’t seem to need to Do this. 

Later, we also need to consider the log adaptation of postgres. Its argValueList.size() == 1 situation is different from that of bigquery and mysql. At that time, I only need method.getname to judge and align for processing. I thought so

Reply:
That sounds fine but why are you extending MethodImplementor and not AbstractRexCallImplementor

Reply:
@tanclary Hello，I have tried it before. Using AbstractRexCallImplementor means that I need to pass a parameter to determine whether the log function belongs to mysql or bigquery. I can completely locate it through method and method name. This does not require me to make a judgment (because method name can be directly mapped to Sqlfunctions method), I think MethodImplementor is more suitable in terms of code size and simplicity, thank you